scala/src/main/scala/akka/contrib/persistence/mongodb/ScalaDriverPersistenceJournaller.scala

```diff
@@ -71,16 +70,16 @@ class ScalaDriverPersistenceJournaller(val driver: ScalaMongoDriver) extends Mon
   }
 
   private[this] def doBatchAppend(writes: immutable.Seq[AtomicWrite], collection: driver.C)(implicit ec: ExecutionContext): Future[immutable.Seq[Try[Unit]]] = {
-    val batch = writes.map(aw => Try(driver.serializeJournal(Atom[Document](aw, driver.useLegacySerialization))))
+    val batch = writes.map(aw => Try(driver.serializeJournal(Atom[BsonValue](aw, driver.useLegacySerialization))))
```

Comment:
Oh, I see what you mean with the changes - I tried making all of the changes, but I fail at this point then.
When making this instead:
```
val batch = writes.map(aw => Try(driver.serializeJournal(Atom[BsonDocument](aw, driver.useLegacySerialization))))
```
The compiler prints following errors:
```
Error:(73, 66) could not find implicit value for parameter dt: akka.contrib.persistence.mongodb.DocumentType[D]
    val batch = writes.map(aw => Try(driver.serializeJournal(Atom(aw, driver.useLegacySerialization))))

Error:(73, 66) not enough arguments for method apply: (implicit ser: akka.serialization.Serialization, implicit ev: Manifest[Nothing], implicit dt: akka.contrib.persistence.mongodb.DocumentType[Nothing], implicit loadClass: akka.contrib.persistence.mongodb.LoadClass)akka.contrib.persistence.mongodb.Atom in object Atom.
Unspecified value parameters dt, loadClass.
    val batch = writes.map(aw => Try(driver.serializeJournal(Atom(aw, driver.useLegacySerialization))))
```

In fact I first tried using `D = BsonDocument` but failed as well at this point.
If you have an idea how to solve this it would be great if you could take over with the adjustments.

Good news for me is that the majority if the changes in the PR are good :)

Reply:
I wasn't thinking about that `DocumentType` constraint, will see what's possible in there, maybe all of these changes have to stay.